eventuals/scheduler.h

```diff
@@ -37,12 +37,44 @@ class Scheduler {
       return previous;
     }
 
-    Context(Scheduler* scheduler)
-      : scheduler_(scheduler) {}
+    auto* scheduler() {
+      return scheduler_;
+    }
 
-    Context(Context&& that) = delete;
+    const std::string& name() {
+      return name_;
+    }
+
+    Context(
+        Scheduler* scheduler,
+        const std::string& name = "default",
+        void* data = nullptr,
+        Context* parent = nullptr)
+      : data_(data),
+        parent_(parent),
+        scheduler_(scheduler),
+        name_(name) {}
+
+    Context(Context&& that)
+      : data_(that.data_),
+        parent_(that.parent_),
+        scheduler_(that.scheduler()),
+        name_(that.name()) {
+      // NOTE: should only get moved before it's "started".
+      CHECK(!that.blocked_ && !callback_ && next_ == nullptr);
```

Comment:
Glad that you caught this!

Let's actually not support move construction at all! In fact, let's set the move constructor to `= delete;`. Then let's remove the `Scheduler::Context::Clone()` and instead have the `Context()` constructor be sufficient for doing a clone and inside the constructor call `scheduler_->Clone(this);`.